E0005 needs to be updated to new format

From: src/test/compile-fail/E0005.rs

E0005 needs a span_label, taking it from:

```
error[E0005]: refutable pattern in local binding: `None` not covered
  --> src/test/compile-fail/E0005.rs:13:9
   |
13 |     let Some(y) = x; //~ ERROR E0005
   |         ^^^^^^^
```

to:
```
error[E0005]: refutable pattern in local binding: `None` not covered
  --> src/test/compile-fail/E0005.rs:13:9
   |
13 |     let Some(y) = x; //~ ERROR E0005
   |         ^^^^^^^ pattern `None` not covered
```

--- a/src/librustc_const_eval/check_match.rs
+++ b/src/librustc_const_eval/check_match.rs
@@ -1073,11 +1073,12 @@ fn check_irrefutable(cx: &MatchCheckCtxt, pat: &Pat, is_fn_arg: bool) {
     };
 
     is_refutable(cx, pat, |uncovered_pat| {
-        span_err!(cx.tcx.sess, pat.span, E0005,
+        let pattern_string = pat_to_string(uncovered_pat);
+        struct_span_err!(cx.tcx.sess, pat.span, E0005,
             "refutable pattern in {}: `{}` not covered",
             origin,
-            pat_to_string(uncovered_pat),
-        );
+            pattern_string,
+        ).span_label(pat.span, &format!("pattern `{}` not covered", pattern_string)).emit();
     });
 }
 

--- a/src/test/compile-fail/E0005.rs
+++ b/src/test/compile-fail/E0005.rs
@@ -11,4 +11,5 @@
 fn main() {
     let x = Some(1);
     let Some(y) = x; //~ ERROR E0005
+    //~| NOTE pattern `None` not covered
 }

